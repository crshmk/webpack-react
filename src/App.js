import React from 'react'
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom'

import './index.css'

import FadeIn from '@components/FadeIn'
import Nav from '@components/Nav'
import Providers from '@store/Providers'
import Routes from './Routes'
import ScrollToTop from './ScrollToTop'

const View = () => (
  <FadeIn>
    <div className="view">
      <Nav />
      <Routes />
    </div>
  </FadeIn>
)

export const App = () => (
  <Router>
  <ScrollToTop />
  <Providers>
  <View />
  </Providers>
  </Router>
)

export default App

export const TestApp = () => (
  <MemoryRouter initialEntries={['/']}>
  <Providers>
  <View />
  </Providers>
  </MemoryRouter>
)
