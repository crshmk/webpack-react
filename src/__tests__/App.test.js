import React from 'react'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

import { TestApp } from "../App"

describe('App', () => {
  test('sanity', () => {
    const { asFragment } = render(<TestApp />)
    
    expect(asFragment()).toMatchSnapshot()

    const homeLink = screen.getByText('Home')
    expect(homeLink).toBeTruthy()
  })
})
