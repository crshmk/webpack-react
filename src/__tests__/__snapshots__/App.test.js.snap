// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App sanity 1`] = `
<DocumentFragment>
  <div
    class="fade-in loading"
  >
    <div
      class="view"
    >
      <nav>
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class=""
              href="/dashboard"
            >
              Dashboard
            </a>
          </li>
          <li>
            <a
              class=""
              href="/users"
            >
              Users
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
